### PR TITLE
[Improvement](decimalv3) Add a config to check overflow for DECIMALV3

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.35 seconds
 load time: 654 seconds
 storage size: 17123669837 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221229055016_clickbench_pr_70678.html

File: fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java [hunk 1]
```diff
@@ -2036,6 +2038,27 @@ private boolean customNullableAlgorithm() {
         if (fn.functionName().equalsIgnoreCase("concat_ws")) {
             return children.get(0).isNullable();
         }
+        if (fn.functionName().equalsIgnoreCase(Operator.MULTIPLY.getName())
+                && fn.getReturnType().isDecimalV3()) {
+            if ((children.get(0).getType().getPrecision() + children.get(1).getType().getPrecision()
+                    > ScalarType.MAX_PRECISION || !children.get(0).getType().isDecimalV3()
+                    || !children.get(1).getType().isDecimalV3()) && ConnectContext.get() != null
+                    && ConnectContext.get().getSessionVariable().checkOverflowForDecimal()) {
+                return true;
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/Expr.java:2047]: make sure the nullable type return same as BE，maybe coredump here

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**